### PR TITLE
Added code to generate event builders for type-safe filtering of events

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds code generation to create event builder types with named and typed topic filter methods.

### Test Plan

Example uses new generated API to filter `from` addresses. Added a couple unit tests.

### CODE REVIEW

File: generate/src/contract/events.rs [hunk 1]
```diff
@@ -190,17 +190,25 @@ fn expand_data_tuple(
 /// Expands into an `Events` type with method definitions for creating event
 /// streams for all non-anonymous contract events in the ABI.
 fn expand_filters(cx: &Context) -> Result<TokenStream> {
-    let filters = cx
+    let anonymous_events = cx
```
Comment (fleupold) [inline generate/src/contract/events.rs]: Isn't the filter logic inverted? it looks like you are getting only events that are not anonymous.

File: examples/events.rs [hunk 2]
```diff
@@ -29,9 +29,8 @@ async fn run() {
     let mut transfers = instance
         .events()
         .transfer()
-        .topic0(Topic::This(accounts[0])) // from
-        .stream()
-        .expect("failed to encode topic filters");
+        .from(Topic::This(accounts[0]))
```
Comment (fleupold) [inline examples/events.rs:32]: Getting better looking every day!

**Review by fleupold — APPROVED** [review]
🍻

File: generate/src/contract/events.rs (hunk 1 above)
Comment (nlordell) [reply generate/src/contract/events.rs]: That was an oopsie with the name :stuck_out_tongue:

**Pull request merged**